File: navigation/src/main/java/com/example/navigation/stepsEngine/field/AmountField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;
import com.fasterxml.jackson.annotation.JsonTypeName;

import java.math.BigDecimal;

@JsonTypeName("AMOUNT")
public class AmountField extends FlowStateField {

    @Override
    public BigDecimal get(FlowState flowState) {
        return flowState.getAmount();
    }

    @Override
    public String getFieldName() {
        return null;
    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/AmountField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize
import java.math.BigDecimal

@Parcelize
class AmountField() : Field() {


    override fun getId(): String {
        return FieldName.AMOUNT
    }

    override fun get(flowState: FlowState): Any? {
        return flowState.amount
    }

    override fun set(flowState: FlowState, value: Any?) {
        flowState.amount = value as BigDecimal
    }

}